src/Middleware/ParametersNameDecodingHandler.cs

```diff
@@ -0,0 +1,53 @@
+// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Kiota.Http.HttpClientLibrary.Middleware.Options;
+
+namespace Microsoft.Kiota.Http.HttpClientLibrary.Middleware;
+
+/// <summary>
+/// This handlers decodes special characters in the request query parameters that had to be encoded due to RFC 6570 restrictions names before executing the request.
+/// </summary>
+public class ParametersNameDecodingHandler: DelegatingHandler
```

Comment:
Should we add this to the default Kiota Client factory?

https://github.com/microsoft/kiota-http-dotnet/blob/8f8f759b21817b34de141adf85adb295f2c0e345/src/KiotaClientFactory.cs#L39

Reply:
thanks for the callout, done